aptos-move/aptos-vm-logging/src/lib.rs

```diff
@@ -65,30 +86,53 @@ pub fn init_speculative_logs(num_txns: usize) {
 /// events storage is not initialized or appropriately sized.
 pub fn speculative_log(level: Level, context: &AdapterLogSchema, message: String) {
     let txn_idx = context.get_txn_idx();
-    match &*BUFFERED_LOG_EVENTS.load() {
-        Some(log_events) => {
-            let log_event = VMLogEntry::new(level, context.clone(), message);
-            if let Err(e) = log_events.record(txn_idx, log_event) {
-                alert!("{:?}", e);
-            };
-        },
-        None => {},
-    };
-}
 
-/// Flushes the first num_to_flush logs in the currently stored logs, and swaps the speculative log / event storage with None.
-/// Must be called after block execution is complete (removes the storage from Arc).
-pub fn flush_speculative_logs(num_to_flush: usize) {
-    if let Some(log_events_ptr) = BUFFERED_LOG_EVENTS.swap(None) {
-        match Arc::try_unwrap(log_events_ptr) {
-            Ok(log_events) => log_events.flush(num_to_flush),
-            Err(_) => {
-                alert!("Speculative log storage must be uniquely owned to flush");
+    if !context.speculation_supported() && speculation_disabled() {
```

Comment:
Shouldn' this be ||

Reply:
agh this was the last change to address comment before merging. Thanks for catching!